pool/pool.go

```diff
@@ -33,25 +35,54 @@ var (
 // that uses a postgres database to store the data
 type Pool struct {
 	storage
-	state        stateInterface
-	l2BridgeAddr common.Address
-	chainID      uint64
-	cfg          Config
-	minGasPrice  *big.Int
+	state                   stateInterface
+	l2BridgeAddr            common.Address
+	chainID                 uint64
+	cfg                     Config
+	minSuggestedGasPrice    *big.Int
+	minSuggestedGasPriceMux *sync.RWMutex
 }
 
 // NewPool creates and initializes an instance of Pool
 func NewPool(cfg Config, s storage, st stateInterface, l2BridgeAddr common.Address, chainID uint64) *Pool {
 	return &Pool{
-		cfg:          cfg,
-		storage:      s,
-		state:        st,
-		l2BridgeAddr: l2BridgeAddr,
-		chainID:      chainID,
-		minGasPrice:  big.NewInt(0).SetUint64(cfg.MinGasPrice),
+		cfg:                     cfg,
+		storage:                 s,
+		state:                   st,
+		l2BridgeAddr:            l2BridgeAddr,
+		chainID:                 chainID,
+		minSuggestedGasPriceMux: new(sync.RWMutex),
 	}
 }
 
+// StartPollingMinSuggestedGasPrice starts polling the minimum suggested gas price
+func (p *Pool) StartPollingMinSuggestedGasPrice(ctx context.Context) {
+	err := p.pollMinSuggestedGasPrice(ctx)
+	if err != nil && err != state.ErrNotFound {
+		log.Fatalf("Error polling min suggested gas price: %v", err)
+	}
```

Comment:
this code is not needed

Reply:
It is because we can't start if there is problem with the DB and we don't have value here. Because we will have later on nil pointer reference panic.